ksqldb-common/src/main/java/io/confluent/ksql/reactive/BaseSubscriber.java

```diff
@@ -147,7 +147,7 @@ protected final void checkContext() {
     VertxUtils.checkContext(context);
   }
 
-  private void runOnRightContext(final Runnable runnable) {
+  protected void runOnRightContext(final Runnable runnable) {
```

Comment:
I'm a little confused, the first line in `makeRequest` checks the context. if that doesn't ensure our property, then what does it do? 

is there any reason not to just _always_ call `makeRequest` wrapped in a `runOnRightContext`? If that's the case could we keep this private and just make `makeRequest` do that?

Reply:
> I'm a little confused, the first line in makeRequest checks the context. if that doesn't ensure our property, then what does it do?

The call to `checkContext` just asserts that we're on the context and is the basis of the exceptions we've seen because we're not in fact on it in some call paths.